CommonTools/RecoAlgos/plugins/JetDeltaRValueMapProducer.cc

```diff
@@ -166,7 +166,7 @@ class JetDeltaRValueMapProducer : public edm::EDProducer {
   const std::vector<std::string>                   valueLabels_;
   const bool                                       lazyParser_;
   bool                                             multiValue_;
-  std::map<std::string, StringObjectFunction<T> >  evaluationMap_;
+  std::map<std::string, std::unique_ptr<const StringObjectFunction<T> > >  evaluationMap_;
```

Comment:
I do not believe the StringObjectFunction is thread safe. Especially if lazyParser_ is true.